polyply/src/annotate_dna.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from random import sample
+from difflib import SequenceMatcher
+from itertools import permutations
+import numpy as np
+import networkx as nx
+from .processor import Processor
+
+base_library = {"DA": "DT", "DT": "DA", "DG": "DC", "DC": "DG",
+                "DA5": "DT5", "DT5": "DA5", "DG5": "DC5", "DC5": "DG5",
+                "DA3": "DT3", "DT3": "DA3", "DG3": "DC3", "DC3": "DG3"
+                }
+
+class AnnotateDNA(Processor):
+    """
+    Given a topology and list indexes that indicate the molecules that
+    contain DNA, the processor generates the appropriate meta_molecule
+    structure. The DNA strands are identified using a greedy search method
+    on the meta_molecule network and later combined with their complementary
+    strand. The generated meta_molecule is later used for both building the
+    DNA structure and backmapping to the desired forcefield.
+    """
+
+    def __init__(self, topology, includes_DNA):
+        """
+        Initalize the processor with a :class:`topology`
+        and a list of DNA molecules in this topology.
+
+        Parameters
+        ------------
+        topology: :class: `polyply.src.topology.Topology`
+        includes_DNA: list[int]
+            node keys list of the meta_molecules tagged as indluding DNA
+        """
+        self.topology = topology
+        self.includes_DNA = [float(ndx) for ndx in includes_DNA]
+
+    def _find_dna_strands(self, meta_molecule):
+        """
+        Extract the subgraphs of meta_molecule that are DNA strands
+
+        Parameters
+        ------------
+        meta_molecule: :class:`polyply.src.MetaMolecule`
+
+        Returns
+        ---------
+        strands: list[:class: `networkx.Graph`]
+            list of DNA strands represented as subgraphs of meta_molecule
+        """
+
+        def _is_nucleobase(node):
+            return (meta_molecule.nodes[node]['resname'] in base_library)
```

Comment:
why the braces around is expression? You don't need them, it will just return the bool. Also this function can be useful elsewhere so I'd move it one level up and give it a proper doc-string